### PR TITLE
[proxy] Wildcard certificate support

### PR DESCRIPTION
**Pull request by pascal-fischer**
## Describe your changes
With this change it is not possible to pass a directory path for wildcard certificates to the proxy with `NB_PROXY_WILDCARD_CERT_DIR` or `--wildcard-cert-dir`.

The proxy will read all the certificates in that folder and cache for the domains matching SNI.

### Checklist
- [ ] Is it a bug fix
- [ ] Is a typo/documentation fix
- [x] Is a feature enhancement
- [ ] It is a refactor
- [ ] Created tests that fail without the change (if possible)

> By submitting this pull request, you confirm that you have read and agree to the terms of the [Contributor License Agreement](https://github.com/netbirdio/netbird/blob/main/CONTRIBUTOR_LICENSE_AGREEMENT.md).

## Documentation
Select exactly one:

- [ ] I added/updated documentation for this change
- [x] Documentation is **not needed** for this change (explain why)

### Docs PR URL (required if "docs added" is checked)
Paste the PR link from https://github.com/netbirdio/docs here:

https://github.com/netbirdio/docs/pull/__

## Summary by CodeRabbit

* **New Features**
  * New CLI flag to point the server at a directory of wildcard TLS certificate pairs; matching wildcard certificates are loaded and served immediately.
  * Wildcard domains served from provided certificates bypass external provisioning when applicable.
  * Background monitoring reloads updated wildcard certificates without restarting the server.

* **Tests**
  * Expanded coverage for wildcard parsing, matching, loading, serving, and related edge cases.

### CODE REVIEW

File: proxy/internal/acme/manager_test.go [hunk 1]
```diff
@@ -100,3 +112,193 @@ func TestDomainStates(t *testing.T) {
...
+
+func TestParseWildcard(t *testing.T) {
+	tests := []struct {
+		pattern    string
+		wantSuffix string
+		wantOK     bool
+	}{
+		{"*.example.com", ".example.com", true},
+		{"*.foo.example.com", ".foo.example.com", true},
+		{"*.COM", ".com", false},      // single-label TLD
```
Comment (lixmal) [inline proxy/internal/acme/manager_test.go]: Do we need to reject this? Could be valid in a private domain context

File: proxy/internal/acme/manager.go [hunk 2]
```diff
@@ -60,54 +91,202 @@ type Manager struct {
...
+	for i := range mgr.wildcards {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			mgr.wildcards[i].watcher.Watch(ctx)
+		}()
+	}
+	wg.Wait()
+}
+
```
Comment (lixmal) [inline proxy/internal/acme/manager.go:172]: This will launch multiple goroutines for the same file if there are multiple SANs in a file

File: proxy/internal/acme/manager.go [hunk 3]
```diff
@@ -60,54 +91,202 @@ type Manager struct {
...
+			continue
+		}
+		// Single-level match: prefix before suffix must have no dots.
+		prefix := strings.TrimSuffix(host, e.suffix)
+		if len(prefix) > 0 && !strings.Contains(prefix, ".") {
+			return e.watcher
+		}
+	}
+	return nil
+}
```
Comment (lixmal) [inline proxy/internal/acme/manager.go]: Dupe of `findWildcardPattern`?
